server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -0,0 +1,237 @@
+                public String executor() {
+                    return ThreadPool.Names.SAME;
+                }
+
+                @Override
+                public AddVotingConfigExclusionsResponse read(StreamInput in) throws IOException {
+                    return new AddVotingConfigExclusionsResponse(in);
+                }
+            }
+        );
+    }
+
+    public void clearVotingConfigExclusion(ActionListener<Void> listener) {
+        final ClearVotingConfigExclusionsRequest clearVotingConfigExclusionsRequest = new ClearVotingConfigExclusionsRequest();
+        transportService.sendRequest(
+            transportService.getLocalNode(),
+            ClearVotingConfigExclusionsAction.NAME,
+            clearVotingConfigExclusionsRequest,
+            new TransportResponseHandler<ClearVotingConfigExclusionsResponse>() {
+                @Override
+                public void handleResponse(ClearVotingConfigExclusionsResponse response) {
+                    listener.onResponse(null);
+                }
+
+                @Override
+                public void handleException(TransportException exp) {
+                    listener.onFailure(exp);
+                }
+
+                @Override
+                public String executor() {
+                    return ThreadPool.Names.SAME;
+                }
+
+                @Override
+                public ClearVotingConfigExclusionsResponse read(StreamInput in) throws IOException {
+                    return new ClearVotingConfigExclusionsResponse(in);
+                }
+            }
+        );
+    }
+
+    public void removeDecommissionedNodes(
+        Set<DiscoveryNode> nodesToBeDecommissioned,
+        String reason,
+        TimeValue timeout,
+        ActionListener<Void> nodesRemovedListener
+    ) {
+        final Map<NodeRemovalClusterStateTaskExecutor.Task, ClusterStateTaskListener> nodesDecommissionTasks = new LinkedHashMap<>();
+        nodesToBeDecommissioned.forEach(discoveryNode -> {
+            final NodeRemovalClusterStateTaskExecutor.Task task = new NodeRemovalClusterStateTaskExecutor.Task(discoveryNode, reason);
+            nodesDecommissionTasks.put(task, nodeRemovalExecutor);
+        });
+        clusterService.submitStateUpdateTasks(
+            "node-decommissioned",
+            nodesDecommissionTasks,
+            ClusterStateTaskConfig.build(Priority.URGENT),
+            nodeRemovalExecutor
+        );
+
+        Predicate<ClusterState> allDecommissionedNodesRemovedPredicate = clusterState -> {
+            Iterator<DiscoveryNode> nodesIter = clusterState.nodes().getNodes().valuesIt();
+            while (nodesIter.hasNext()) {
+                final DiscoveryNode node = nodesIter.next();
+                // check if the node is part of node decommissioned list
+                if (nodesToBeDecommissioned.contains(node)) {
+                    return false;
+                }
+            }
+            return true;
+        };
```

Comment:
Can we use `Sets.intersection(first, second);`

Reply:
Updating the logic to use retain all for intersecttion - 
```
Predicate<ClusterState> allDecommissionedNodesRemovedPredicate = clusterState -> {
            Set<DiscoveryNode> intersection = Arrays.stream(
                clusterState.nodes().getNodes().values().toArray(DiscoveryNode.class)).collect(Collectors.toSet());
            intersection.retainAll(nodesToBeDecommissioned);
            return intersection.size() == 0;
        };
```